### PR TITLE
Dataflow: Remove src/sink grouping feature

### PR DESCRIPTION
**Pull request by aschackmull**
This hasn't proven useful, and a quick poll showed no known usage.

### CODE REVIEW

**Review by hvitved — APPROVED** [review]
LGTM (assuming the Python test failures are unrelated).

**Comment by aschackmull** [comment]
> LGTM (assuming the Python test failures are unrelated).

It was. It passed on a retry.

File: shared/dataflow/codeql/dataflow/internal/DataFlowImpl.qll [hunk 1]
```diff
@@ -4284,11 +4259,19 @@ module MakeImpl<LocationSig Location, InputSig<Location> Lang> {
...
+      deprecated final predicate isSourceGroup(string group) { none() }
 
-      /** Holds if this node is a grouping of sink nodes. */
-      final predicate isSinkGroup(string group) { this = TPathNodeSinkGroup(group) }
+      /**
+       * DEPRECATED: This functionality is no longer available.
+       *
+       * Holds if this node is a grouping of sink nodes.
+       */
+      deprecated final predicate isSinkGroup(string group) { none() }
```
Comment (geoffw0) [inline shared/dataflow/codeql/dataflow/internal/DataFlowImpl.qll:4274]: So to be clear, we're deprecating _and_ removing the feature, the only thing that's still supported is that existing code will still compile (but ignore souce/sink grouping) thanks to these stubs, for the deprecation cycle.

**Review by geoffw0 — APPROVED** [review]
Fine with me (speaking for CPP and Swift).

File: shared/dataflow/codeql/dataflow/internal/DataFlowImpl.qll (hunk 1 above)
Comment (aschackmull) [reply shared/dataflow/codeql/dataflow/internal/DataFlowImpl.qll:4274]: Yes.

**Pull request merged**